docs/content/ecs/component-updates.md

```diff
@@ -49,6 +51,72 @@ public class SendHealthUpdateSystem : ComponentSystem
 }
 ```
 
+#### Component update edge case
+
+There is one edge case that you need to be careful about which is a result of how component changes are tracked and [C# reference type](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/reference-types) semantics.
+
+Simply adding or removing elements to a `List<T>` or a `Dictionary<K, V>` field in a component is not sufficient to trigger a component update. Instead, you need to set the field back to itself to trigger the update. This applies to `List<T>` and `Dictionary<K, V>` fields within types as well.
```

Comment:
I commented suggesting for an example....before I spotted the excellent example - maybe call out the example?

```suggestion
Simply adding or removing elements to a `List<T>` or a `Dictionary<K, V>` field in a component is not sufficient to trigger a component update. Instead, you need to set the field back to itself to trigger the update, as shown in the example below. This applies to `List<T>` and `Dictionary<K, V>` fields within types as well.
```